samples/sensor/bme280/app.overlay

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2020, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * By default, this overlay sets up a single BME280 sensor
+ * on the standard Arduino header I2C pins.
+ *
+ * If your board's devicetree has an "arduino_i2c" node label,
+ * you can use this overlay as-is, wiring the BME280 sensor to the
+ * breakout header accordingly:
+ *
+ * https://upload.wikimedia.org/wikipedia/commons/c/c9/Pinout_of_ARDUINO_Board_and_ATMega328PU.svg
+ *
+ * If you're not sure, just compile the application and see. You'll
+ * get a build error if there is no "arduino_i2c" support.
+ *
+ * You can also change the "#if 1" line to "#if 0" to switch to the
+ * Arduino SPI pins instead.
+ *
+ * If your board does not have an Arduino header, you can create a file
+ * named boards/YOUR_BOARD_NAME.overlay in this sample and set up the
+ * sensor device on an I2C or SPI bus available in your devicetree.
+ * See boards/adafruit_feather_m0_basic_proto.overlay for an example.
+ */
+
+#if 1
```

Comment:
Not fond of the `#if #else` but I don't have better proposal.
For sanitycheck/twister, we could have used sample.yaml to sort this out, but I ack this should be available from west

Reply:
> Not fond of the `#if #else` but I don't have better proposal.

@erwango I don't like it much, either, so I've switched to just defining both I2C and SPI nodes in the devicetree and selecting one via alias. I added a bunch of comments for new users and updated the README as well.